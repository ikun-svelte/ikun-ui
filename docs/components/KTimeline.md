---
title: KTimeline
lang: en-US
---

# KTimeline

Vertical display timeline.

## Install

::: code-group

```bash [pnpm]
pnpm add @ikun-ui/timeline
```

```bash [yarn]
yarn add @ikun-ui/timeline
```

```bash [npm]
npm install @ikun-ui/timeline
```

:::

## Basic usage

Basic timeline.

<demo src="timeline/basic.svelte"  github='Timeline'></demo>

## Color

Set the color of circles.

<demo src="timeline/color.svelte"  github='Timeline'></demo>

## Last node and Reversing

When the timeline is incomplete and ongoing, 
put a ghost node at last. Set `pending` as truthy value to enable displaying pending item.
`reverse={true}` is used for reversing nodes.

<demo src="timeline/reverse.svelte"  github='Timeline'></demo>

## Custom render pending content

You can customize the rendering of pending nodes through the pending slot and pendingDot slot. 

::: tip
* The `pendingDot` slot provides style class `cls`, rendering node item `item` and rendering index `index`  
* The `pending` slot provides the style class `cls`, the content of the rendering node item `children` and the rendering index `index`  
:::

<demo src="timeline/pending.svelte"  github='Timeline'></demo>

## Right alternate

Right alternate timeline.

<demo src="timeline/right.svelte"  github='Timeline'></demo>

## Alternate

Alternate timeline.

<demo src="timeline/alternate.svelte"  github='Timeline'></demo>

## Label

Use `label` show time alone.

<demo src="timeline/label.svelte"  github='Timeline'></demo>

## Custom render

You can customize the rendering of pending nodes through the label slot „ÄÅdot slot and children slot.

::: tip
* The `dot` slot provides style class `cls`, rendering node item `item` and rendering index `index`
* The `children` slot provides the style class `cls`, the content of the rendering node item `children` and the rendering index `index`  
* The `label` slot provides the style class `cls`, the content of the rendering node item `label` and the rendering index `index`  
  :::

<demo src="timeline/custom.svelte"  github='Timeline'></demo>

## Timeline Props

| Name    | Type                           | Default | Description                                                                          |
| ------- | ------------------------------ | ------- | ------------------------------------------------------------------------------------ |
| mode    | `left` \| `alternate`\| `left` | `left`  | By sending `alternate` the timeline will distribute the nodes to the left and right. |
| reverse | `boolean`                      | `false` | Whether reverse nodes or not.                                                        |
| pending | `string` \| `boolean`          | `false` | Set the last ghost node's existence or its content.                                  |
| items   | `KTimelineItem[]`              | `[]`    | Each node of timeline                                                                |
| cls     | `string`                       | `''`    | Additional class for component                                                       |
| attrs   | `any`                          | `{}`    | Additional attributes                                                                |

```typescript
export interface KTimelineItem<L = string, C = string> {
	/**
	 * Set the circle's color
	 */
	color?: string;
	/**
	 * Customize node position
	 */
	position?: 'left' | 'right';
	/**
	 * Set the label
	 */
	label?: L;
	/**
	 * Set the content
	 */
	children: C;
	/**
	 * Uniquely identifies
	 */
	uid: string | number;
}
```

## Timeline Slots

| Name       | Description                                                        |
| ---------- | ------------------------------------------------------------------ |
| dot        | content of the `dot`.                                              |
| pendingDot | content of the `pending dot`. it will only work if pending is true |
| children   | content of the `content`.                                          |
| pending    | content of the `pending`.                                          |
| label      | content of the `label`.                                            |
