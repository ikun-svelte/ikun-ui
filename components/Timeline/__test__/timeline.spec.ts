import { afterEach, beforeEach, describe, expect, test, vi } from 'vitest';
import KTimeline from '../src';
import KTBasic from './fixture/basic.svelte';
import KTColor from './fixture/color.svelte';
import KTPosModeLeft from './fixture/position.mode.left.svelte';
import KTPosModeRight from './fixture/position.mode.right.svelte';
import KTPosModeAlternate from './fixture/position.mode.alternate.svelte';
import KTLabelModeLeft from './fixture/label.mode.left.svelte';
import KTLabelModeRight from './fixture/label.mode.right.svelte';
import KTLabelModeAlternate from './fixture/label.mode.alternate.svelte';
import KTChildrenModeLeft from './fixture/children.mode.left.svelte';
import KTChildrenModeRight from './fixture/children.mode.right.svelte';
import KTChildrenModeAlternate from './fixture/children.mode.alternate.svelte';
import KTSlotsLabel from './fixture/slots.label.svelte';
import KTSlotsChildren from './fixture/slots.children.svelte';
import KTSlotsDot from './fixture/slots.dot.svelte';
import KTSlotsPending from './fixture/slots.pending.svelte';
import KTSlotsPendingDot from './fixture/slots.pending-dot.svelte';
import KTReverse from './fixture/reverse.svelte';
import KTReversePending from './fixture/reverse.pending.svelte';
import KTPending from './fixture/pending.true.svelte';
import KTPendingStr from './fixture/pending.string.svelte';
import KTMode from './fixture/mode.svelte';
import KTModeLabel from './fixture/mode.label.svelte';
let host;
const initHost = () => {
	host = globalThis.document.createElement('div');
	host.setAttribute('id', 'host');
	globalThis.document.body.appendChild(host);
};
beforeEach(() => {
	initHost();
	vi.useFakeTimers();
});
afterEach(() => {
	host.remove();
	vi.useRealTimers();
});

describe('Test: KTimeline', () => {
	test('basic render', async () => {
		// @ts-ignore
		const instance = new KTBasic({
			target: host
		});
		expect(instance).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});

	test('props: cls', async () => {
		// @ts-ignore
		const instance = new KTimeline({
			target: host,
			props: {
				cls: 'k-timeline--test'
			}
		});
		expect(instance).toBeTruthy();
		expect(host!.innerHTML.includes('k-timeline--test')).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});

	test('props: pending is true', async () => {
		// @ts-ignore
		const instance = new KTPending({
			target: host
		});
		expect(instance).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});

	test('props: pending is string', async () => {
		// @ts-ignore
		const instance = new KTPendingStr({
			target: host
		});
		expect(instance).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});

	test('props: reverse', async () => {
		// @ts-ignore
		const instance = new KTReverse({
			target: host
		});
		expect(instance).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});

	test('props: mode', async () => {
		// @ts-ignore
		const instance = new KTMode({
			target: host
		});
		expect(instance).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});

	test('props: mode & label', async () => {
		// @ts-ignore
		const instance = new KTModeLabel({
			target: host
		});
		expect(instance).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});

	test('props: reverse & pending', async () => {
		// @ts-ignore
		const instance = new KTReversePending({
			target: host
		});
		expect(instance).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});

	test('KTimelineItem: color', async () => {
		// @ts-ignore
		const instance = new KTColor({
			target: host
		});
		expect(instance).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});

	test('KTimelineItem: position', async () => {
		// @ts-ignore
		new KTPosModeLeft({
			target: host
		});

		// @ts-ignore
		new KTPosModeRight({
			target: host
		});

		// @ts-ignore
		new KTPosModeAlternate({
			target: host
		});
		expect(host.innerHTML).matchSnapshot();
	});

	test('KTimelineItem: label', async () => {
		// @ts-ignore
		new KTLabelModeLeft({
			target: host
		});

		// @ts-ignore
		new KTLabelModeRight({
			target: host
		});

		// @ts-ignore
		new KTLabelModeAlternate({
			target: host
		});
		expect(host.innerHTML).matchSnapshot();
	});

	test('KTimelineItem: children', async () => {
		// @ts-ignore
		new KTChildrenModeLeft({
			target: host
		});

		// @ts-ignore
		new KTChildrenModeRight({
			target: host
		});

		// @ts-ignore
		new KTChildrenModeAlternate({
			target: host
		});
		expect(host.innerHTML).matchSnapshot();
	});

	test('slots: label', async () => {
		// @ts-ignore
		new KTSlotsLabel({
			target: host
		});
		expect(host.innerHTML).matchSnapshot();
	});

	test('slots: children', async () => {
		// @ts-ignore
		new KTSlotsChildren({
			target: host
		});
		expect(host.innerHTML).matchSnapshot();
	});

	test('slots: dot', async () => {
		// @ts-ignore
		new KTSlotsDot({
			target: host
		});
		expect(host.innerHTML).matchSnapshot();
	});

	test('slots: pending', async () => {
		// @ts-ignore
		new KTSlotsPending({
			target: host
		});
		expect(host.innerHTML).matchSnapshot();
	});

	test('slots: pendingDot', async () => {
		// @ts-ignore
		new KTSlotsPendingDot({
			target: host
		});
		expect(host.innerHTML).matchSnapshot();
	});
});
