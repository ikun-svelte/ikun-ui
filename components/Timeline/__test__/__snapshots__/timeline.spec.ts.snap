// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test: KTimeline > KTimelineItem: children 1`] = `"<ul class="k-timeline"><li class="k-timeline-item"><div class="k-timeline-item-tail" style="height: calc(100% - 10px);"></div>  <div class="k-timeline-item-head k-timeline-item-head--left"></div>  <div class="k-timeline-item-content k-timeline-item-content--dark">Technical testing 2015-09-01</div>   </li><li class="k-timeline-item">  <div class="k-timeline-item-head k-timeline-item-head--left"></div>  <div class="k-timeline-item-content k-timeline-item-content--dark">Network problems being solved</div>   </li></ul><ul class="k-timeline"><li class="k-timeline-item"><div class="k-timeline-item-tail k-timeline-item-tail--right" style="height: calc(100% - 10px);"></div>  <div class="k-timeline-item-head k-timeline-item-head--right"></div>  <div class="k-timeline-item-content k-timeline-item-content--right k-timeline-item-content--dark">Technical testing 2015-09-01</div>   </li><li class="k-timeline-item">  <div class="k-timeline-item-head k-timeline-item-head--right"></div>  <div class="k-timeline-item-content k-timeline-item-content--right k-timeline-item-content--dark">Network problems being solved</div>   </li></ul><ul class="k-timeline"><li class="k-timeline-item"><div class="k-timeline-item-tail k-timeline-item-tail--center" style="height: calc(100% - 10px);"></div>  <div class="k-timeline-item-head k-timeline-item-head--center"></div>  <div class="k-timeline-item-content k-timeline-item-content--cr k-timeline-item-content--dark">Technical testing 2015-09-01</div>   </li><li class="k-timeline-item">  <div class="k-timeline-item-head k-timeline-item-head--center"></div>  <div class="k-timeline-item-content k-timeline-item-content--cl k-timeline-item-content--dark">Network problems being solved</div>   </li></ul>"`;

exports[`Test: KTimeline > KTimelineItem: color 1`] = `"<ul class="k-timeline"><li class="k-timeline-item"><div class="k-timeline-item-tail" style="height: calc(100% - 10px);"></div>  <div class="k-timeline-item-head k-timeline-item-head--left" style="color: rgb(0, 255, 255); border-color: #00ffff;"></div>  <div class="k-timeline-item-content k-timeline-item-content--dark">Technical testing 2015-09-01</div>   </li><li class="k-timeline-item">  <div class="k-timeline-item-head k-timeline-item-head--left" style="color: rgb(0, 255, 255); border-color: #00ffff;"></div>  <div class="k-timeline-item-content k-timeline-item-content--dark">Network problems being solved</div>   </li></ul>"`;

exports[`Test: KTimeline > KTimelineItem: label 1`] = `"<ul class="k-timeline"><li class="k-timeline-item"><div class="k-timeline-item-tail k-timeline-item-tail--center" style="height: calc(100% - 10px);"></div>  <div class="k-timeline-item-head k-timeline-item-head--center"></div>  <div class="k-timeline-item-content k-timeline-item-content--cl k-timeline-item-content--dark">Technical testing 2015-09-01</div>  <div class="k-timeline-item-label k-timeline-item-label--cr k-timeline-item-label--dark">ikun ui 1</div> </li><li class="k-timeline-item">  <div class="k-timeline-item-head k-timeline-item-head--center"></div>  <div class="k-timeline-item-content k-timeline-item-content--cl k-timeline-item-content--dark">Network problems being solved</div>  <div class="k-timeline-item-label k-timeline-item-label--cr k-timeline-item-label--dark">ikun ui 2</div> </li></ul><ul class="k-timeline"><li class="k-timeline-item"><div class="k-timeline-item-tail k-timeline-item-tail--center" style="height: calc(100% - 10px);"></div>  <div class="k-timeline-item-head k-timeline-item-head--center"></div>  <div class="k-timeline-item-content k-timeline-item-content--cr k-timeline-item-content--dark">Technical testing 2015-09-01</div>  <div class="k-timeline-item-label k-timeline-item-label--cl k-timeline-item-label--dark">ikun ui 1</div> </li><li class="k-timeline-item">  <div class="k-timeline-item-head k-timeline-item-head--center"></div>  <div class="k-timeline-item-content k-timeline-item-content--cr k-timeline-item-content--dark">Network problems being solved</div>  <div class="k-timeline-item-label k-timeline-item-label--cl k-timeline-item-label--dark">ikun ui 2</div> </li></ul><ul class="k-timeline"><li class="k-timeline-item"><div class="k-timeline-item-tail k-timeline-item-tail--center" style="height: calc(100% - 10px);"></div>  <div class="k-timeline-item-head k-timeline-item-head--center"></div>  <div class="k-timeline-item-content k-timeline-item-content--cr k-timeline-item-content--dark">Technical testing 2015-09-01</div>  <div class="k-timeline-item-label k-timeline-item-label--cl k-timeline-item-label--dark">ikun ui 1</div> </li><li class="k-timeline-item">  <div class="k-timeline-item-head k-timeline-item-head--center"></div>  <div class="k-timeline-item-content k-timeline-item-content--cl k-timeline-item-content--dark">Network problems being solved</div>  <div class="k-timeline-item-label k-timeline-item-label--cr k-timeline-item-label--dark">ikun ui 2</div> </li></ul>"`;

exports[`Test: KTimeline > KTimelineItem: position 1`] = `"<ul class="k-timeline"><li class="k-timeline-item"><div class="k-timeline-item-tail" style="height: calc(100% - 10px);"></div>  <div class="k-timeline-item-head k-timeline-item-head--left"></div>  <div class="k-timeline-item-content k-timeline-item-content--dark">Technical testing 2015-09-01</div>   </li><li class="k-timeline-item">  <div class="k-timeline-item-head k-timeline-item-head--left"></div>  <div class="k-timeline-item-content k-timeline-item-content--dark">Network problems being solved</div>   </li></ul><ul class="k-timeline"><li class="k-timeline-item"><div class="k-timeline-item-tail k-timeline-item-tail--right" style="height: calc(100% - 10px);"></div>  <div class="k-timeline-item-head k-timeline-item-head--right"></div>  <div class="k-timeline-item-content k-timeline-item-content--right k-timeline-item-content--dark">Technical testing 2015-09-01</div>   </li><li class="k-timeline-item">  <div class="k-timeline-item-head k-timeline-item-head--right"></div>  <div class="k-timeline-item-content k-timeline-item-content--right k-timeline-item-content--dark">Network problems being solved</div>   </li></ul><ul class="k-timeline"><li class="k-timeline-item"><div class="k-timeline-item-tail k-timeline-item-tail--center" style="height: calc(100% - 10px);"></div>  <div class="k-timeline-item-head k-timeline-item-head--center"></div>  <div class="k-timeline-item-content k-timeline-item-content--cr k-timeline-item-c__right k-timeline-item-content--dark">Technical testing 2015-09-01</div>   </li><li class="k-timeline-item">  <div class="k-timeline-item-head k-timeline-item-head--center"></div>  <div class="k-timeline-item-content k-timeline-item-content--cl k-timeline-item-c__left k-timeline-item-content--dark">Network problems being solved</div>   </li></ul>"`;

exports[`Test: KTimeline > basic render 1`] = `"<ul class="k-timeline"><li class="k-timeline-item"><div class="k-timeline-item-tail" style="height: calc(100% - 10px);"></div>  <div class="k-timeline-item-head k-timeline-item-head--left"></div>  <div class="k-timeline-item-content k-timeline-item-content--dark">Technical testing 2015-09-01</div>   </li><li class="k-timeline-item">  <div class="k-timeline-item-head k-timeline-item-head--left"></div>  <div class="k-timeline-item-content k-timeline-item-content--dark">Network problems being solved</div>   </li></ul>"`;

exports[`Test: KTimeline > props: cls 1`] = `"<ul class="k-timeline k-timeline--test"></ul>"`;

exports[`Test: KTimeline > props: mode & label 1`] = `"<ul class="k-timeline"><li class="k-timeline-item"><div class="k-timeline-item-tail k-timeline-item-tail--center" style="height: calc(100% - 10px);"></div>  <div class="k-timeline-item-head k-timeline-item-head--center"></div>  <div class="k-timeline-item-content k-timeline-item-content--cl k-timeline-item-content--dark">Technical testing 2015-09-01</div>  <div class="k-timeline-item-label k-timeline-item-label--cr k-timeline-item-label--dark">ikun ui 1</div> </li><li class="k-timeline-item">  <div class="k-timeline-item-head k-timeline-item-head--center"></div>  <div class="k-timeline-item-content k-timeline-item-content--cl k-timeline-item-content--dark">Network problems being solved</div>  <div class="k-timeline-item-label k-timeline-item-label--cr k-timeline-item-label--dark">ikun ui 2</div> </li></ul> <ul class="k-timeline"><li class="k-timeline-item"><div class="k-timeline-item-tail k-timeline-item-tail--center" style="height: calc(100% - 10px);"></div>  <div class="k-timeline-item-head k-timeline-item-head--center"></div>  <div class="k-timeline-item-content k-timeline-item-content--cr k-timeline-item-content--dark">Technical testing 2015-09-01</div>  <div class="k-timeline-item-label k-timeline-item-label--cl k-timeline-item-label--dark">ikun ui 1</div> </li><li class="k-timeline-item">  <div class="k-timeline-item-head k-timeline-item-head--center"></div>  <div class="k-timeline-item-content k-timeline-item-content--cl k-timeline-item-content--dark">Network problems being solved</div>  <div class="k-timeline-item-label k-timeline-item-label--cr k-timeline-item-label--dark">ikun ui 2</div> </li></ul> <ul class="k-timeline"><li class="k-timeline-item"><div class="k-timeline-item-tail k-timeline-item-tail--center" style="height: calc(100% - 10px);"></div>  <div class="k-timeline-item-head k-timeline-item-head--center"></div>  <div class="k-timeline-item-content k-timeline-item-content--cr k-timeline-item-content--dark">Technical testing 2015-09-01</div>  <div class="k-timeline-item-label k-timeline-item-label--cl k-timeline-item-label--dark">ikun ui 1</div> </li><li class="k-timeline-item">  <div class="k-timeline-item-head k-timeline-item-head--center"></div>  <div class="k-timeline-item-content k-timeline-item-content--cr k-timeline-item-content--dark">Network problems being solved</div>  <div class="k-timeline-item-label k-timeline-item-label--cl k-timeline-item-label--dark">ikun ui 2</div> </li></ul>"`;

exports[`Test: KTimeline > props: mode 1`] = `"<ul class="k-timeline"><li class="k-timeline-item"><div class="k-timeline-item-tail" style="height: calc(100% - 10px);"></div>  <div class="k-timeline-item-head k-timeline-item-head--left"></div>  <div class="k-timeline-item-content k-timeline-item-content--dark">Technical testing 2015-09-01</div>   </li><li class="k-timeline-item">  <div class="k-timeline-item-head k-timeline-item-head--left"></div>  <div class="k-timeline-item-content k-timeline-item-content--dark">Network problems being solved</div>   </li></ul> <ul class="k-timeline"><li class="k-timeline-item"><div class="k-timeline-item-tail k-timeline-item-tail--center" style="height: calc(100% - 10px);"></div>  <div class="k-timeline-item-head k-timeline-item-head--center"></div>  <div class="k-timeline-item-content k-timeline-item-content--cr k-timeline-item-content--dark">Technical testing 2015-09-01</div>   </li><li class="k-timeline-item">  <div class="k-timeline-item-head k-timeline-item-head--center"></div>  <div class="k-timeline-item-content k-timeline-item-content--cl k-timeline-item-content--dark">Network problems being solved</div>   </li></ul> <ul class="k-timeline"><li class="k-timeline-item"><div class="k-timeline-item-tail k-timeline-item-tail--right" style="height: calc(100% - 10px);"></div>  <div class="k-timeline-item-head k-timeline-item-head--right"></div>  <div class="k-timeline-item-content k-timeline-item-content--right k-timeline-item-content--dark">Technical testing 2015-09-01</div>   </li><li class="k-timeline-item">  <div class="k-timeline-item-head k-timeline-item-head--right"></div>  <div class="k-timeline-item-content k-timeline-item-content--right k-timeline-item-content--dark">Network problems being solved</div>   </li></ul>"`;

exports[`Test: KTimeline > props: pending is string 1`] = `"<ul class="k-timeline"><li class="k-timeline-item"> <div class="k-timeline-item-tail k-timeline-item-tail--pending" style="height: calc(100% - 10px);"></div> <div class="k-timeline-item-head k-timeline-item-head--left"></div>  <div class="k-timeline-item-content k-timeline-item-content--dark">Technical testing 2015-09-01</div>   </li><li class="k-timeline-item">   <div class="k-timeline-item-head k-timeline-item-head--left k-timeline-item-head--pending"></div>  <div class="k-timeline-item-content k-timeline-item-content--dark">ikun-ui</div>  </li></ul>"`;

exports[`Test: KTimeline > props: pending is true 1`] = `"<ul class="k-timeline"><li class="k-timeline-item"> <div class="k-timeline-item-tail k-timeline-item-tail--pending" style="height: calc(100% - 10px);"></div> <div class="k-timeline-item-head k-timeline-item-head--left"></div>  <div class="k-timeline-item-content k-timeline-item-content--dark">Technical testing 2015-09-01</div>   </li><li class="k-timeline-item">   <div class="k-timeline-item-head k-timeline-item-head--left k-timeline-item-head--pending"></div>  <div class="k-timeline-item-content k-timeline-item-content--dark">Network problems being solved</div>  </li></ul>"`;

exports[`Test: KTimeline > props: reverse & pending 1`] = `"<ul class="k-timeline"><li class="k-timeline-item"> <div class="k-timeline-item-tail k-timeline-item-tail--pending" style="height: calc(100% - 10px);"></div>  <div class="k-timeline-item-head k-timeline-item-head--left k-timeline-item-head--pending"></div>  <div class="k-timeline-item-content k-timeline-item-content--dark">Network problems being solved</div>  </li><li class="k-timeline-item">  <div class="k-timeline-item-head k-timeline-item-head--left"></div>  <div class="k-timeline-item-content k-timeline-item-content--dark">Technical testing 2015-09-01</div>   </li></ul>"`;

exports[`Test: KTimeline > props: reverse 1`] = `"<ul class="k-timeline"><li class="k-timeline-item"><div class="k-timeline-item-tail" style="height: calc(100% - 10px);"></div>  <div class="k-timeline-item-head k-timeline-item-head--left"></div>  <div class="k-timeline-item-content k-timeline-item-content--dark">Network problems being solved</div>   </li><li class="k-timeline-item">  <div class="k-timeline-item-head k-timeline-item-head--left"></div>  <div class="k-timeline-item-content k-timeline-item-content--dark">Technical testing 2015-09-01</div>   </li></ul>"`;

exports[`Test: KTimeline > slots: children 1`] = `"<ul class="k-timeline"><li class="k-timeline-item"><div class="k-timeline-item-tail" style="height: calc(100% - 10px);"></div>  <div class="k-timeline-item-head k-timeline-item-head--left"></div>  <div slot="children" class="k-timeline-item-content k-timeline-item-content--dark">🍒 Technical testing 2015-09-01</div>   </li><li class="k-timeline-item">  <div class="k-timeline-item-head k-timeline-item-head--left"></div>  <div slot="children" class="k-timeline-item-content k-timeline-item-content--dark">🍒 Network problems being solved</div>   </li></ul>"`;

exports[`Test: KTimeline > slots: dot 1`] = `"<ul class="k-timeline"><li class="k-timeline-item"><div class="k-timeline-item-tail" style="height: calc(100% - 10px);"></div>  <div slot="dot" class="k-timeline-item-head--left">⭕</div>  <div class="k-timeline-item-content k-timeline-item-content--dark">Technical testing 2015-09-01</div>   </li><li class="k-timeline-item">  <div slot="dot" class="k-timeline-item-head--left">⭕</div>  <div class="k-timeline-item-content k-timeline-item-content--dark">Network problems being solved</div>   </li></ul>"`;

exports[`Test: KTimeline > slots: label 1`] = `"<ul class="k-timeline"><li class="k-timeline-item"><div class="k-timeline-item-tail k-timeline-item-tail--center" style="height: calc(100% - 10px);"></div>  <div class="k-timeline-item-head k-timeline-item-head--center"></div>  <div class="k-timeline-item-content k-timeline-item-content--cl k-timeline-item-content--dark">Technical testing 2015-09-01</div>  <div slot="label" class="k-timeline-item-label k-timeline-item-label--cr k-timeline-item-label--dark">🔥 ikun ui 1</div> </li><li class="k-timeline-item">  <div class="k-timeline-item-head k-timeline-item-head--center"></div>  <div class="k-timeline-item-content k-timeline-item-content--cl k-timeline-item-content--dark">Network problems being solved</div>  <div slot="label" class="k-timeline-item-label k-timeline-item-label--cr k-timeline-item-label--dark">🔥 ikun ui 2</div> </li></ul>"`;

exports[`Test: KTimeline > slots: pending 1`] = `"<ul class="k-timeline"><li class="k-timeline-item"> <div class="k-timeline-item-tail k-timeline-item-tail--pending" style="height: calc(100% - 10px);"></div> <div class="k-timeline-item-head k-timeline-item-head--left"></div>  <div class="k-timeline-item-content k-timeline-item-content--dark">Technical testing 2015-09-01</div>   </li><li class="k-timeline-item">   <div class="k-timeline-item-head k-timeline-item-head--left k-timeline-item-head--pending"></div>  <div slot="pending" class="k-timeline-item-content k-timeline-item-content--dark">pending</div>  </li></ul>"`;

exports[`Test: KTimeline > slots: pendingDot 1`] = `"<ul class="k-timeline"><li class="k-timeline-item"> <div class="k-timeline-item-tail k-timeline-item-tail--pending" style="height: calc(100% - 10px);"></div> <div class="k-timeline-item-head k-timeline-item-head--left"></div>  <div class="k-timeline-item-content k-timeline-item-content--dark">Technical testing 2015-09-01</div>   </li><li class="k-timeline-item">   <div slot="pendingDot" class="k-timeline-item-head--left">❌</div>  <div class="k-timeline-item-content k-timeline-item-content--dark">Network problems being solved</div>  </li></ul> <ul class="k-timeline"><li class="k-timeline-item"><div class="k-timeline-item-tail" style="height: calc(100% - 10px);"></div>  <div class="k-timeline-item-head k-timeline-item-head--left"></div>  <div class="k-timeline-item-content k-timeline-item-content--dark">Technical testing 2015-09-01</div>   </li><li class="k-timeline-item">  <div class="k-timeline-item-head k-timeline-item-head--left"></div>  <div class="k-timeline-item-content k-timeline-item-content--dark">Network problems being solved</div>   </li></ul>"`;
